Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -56,6 +56,43 @@ public class ContestsBusinessService : IContestsBusinessService
             this.contestValidationService = contestValidationService;
         }
 
+        public async Task<ContestDetailsServiceModel> GetContestDetails(int id, bool official)
+        {
+            var user = this.userProviderService.GetCurrentUser();
+            var contest = await this.contestsData.GetByIdWithProblems(id);
+
+            var validationResult = this.contestValidationService.GetValidationResult((
+                contest,
+                id,
+                user.Id,
+                user.IsAdmin,
+                official) !);
+
+            if (!validationResult.IsValid)
+            {
+                throw new BusinessServiceException(validationResult.Message);
+            }
+
+            var participant = await this.participantsData
+                .GetWithContestByContestByUserAndIsOfficial(
+                    id,
+                    user.Id!,
+                    official);
+
+            var userIsAdminInContest = user.IsAdmin || IsUserLecturerInContest(contest!, user.Id!);
+            var isOfficialOnlineContest = contest!.CanBeCompeted && contest.IsOnlineExam;
+
+            var contestDetailsServiceModel = contest.Map<ContestDetailsServiceModel>();
+            if (!userIsAdminInContest && isOfficialOnlineContest && participant != null)
+            {
+                var problemsForParticipant = participant.ProblemsForParticipants.Select(x => x.Problem);
+                contestDetailsServiceModel.Problems = problemsForParticipant.Map<ICollection<ContestProblemServiceModel>>();
+                contestDetailsServiceModel.IsUserParticipant = true;
```

Comment:
Yes, actually I need to check only if the contest is online, otherwise the user does not see the results button in a practice contest, even if he/she is a participant in it.

Reply:
Removed the check for online contest as well. User should be able to see the results also for onsite contests.

Reply:
But why did you add `contest!.CanBeCompeted` in the check?

Reply:
Reworked as discussed.